src/Paket.Core/Simplifier.fs

```diff
@@ -34,60 +34,50 @@ let removePackage(packageName, indirectPackages, fileName, interactive) =
-
-let simplifyReferencesFile (refFile, flatLookup, interactive) =
-    let create refFile indirect =
-        let newPackages = 
-            refFile.NugetPackages 
-            |> List.filter (fun p -> not <| removePackage(p, indirect, refFile.FileName, interactive))
-        { refFile with NugetPackages = newPackages }
-
-    let indirect = findIndirect(refFile.NugetPackages, 
-                                flatLookup, 
-                                (fun p -> ReferenceNotFoundInLockFile(refFile.FileName,p)))
-                 
-    create refFile
-    <!> indirect
+    let! indirect = findIndirect(packages, flatLookup, DependencyNotFoundInLockFile)
+
+    let newPackages = 
+        dependenciesFile.Packages
+        |> List.filter (fun package -> not <| removePackage(package.Name, indirect, dependenciesFile.FileName, interactive))
+    let d = dependenciesFile
+    return DependenciesFile(d.FileName, d.Options, d.Sources, newPackages, d.RemoteFiles)
+}
+
+let simplifyReferencesFile (refFile, flatLookup, interactive) = rop {
+    let! indirect = findIndirect(refFile.NugetPackages, 
+                            flatLookup, 
+                            (fun p -> ReferenceNotFoundInLockFile(refFile.FileName,p)))
+
+    let newPackages = 
+        refFile.NugetPackages 
+        |> List.filter (fun p -> not <| removePackage(p, indirect, refFile.FileName, interactive))
+
+    return { refFile with NugetPackages = newPackages }
+}
 
 let beforeAndAfter environment dependenciesFile projects =
         environment,
         { environment with DependenciesFile = dependenciesFile
                            Projects = projects }
 
-let ensureNotInStrictMode environment =
-    if not environment.DependenciesFile.Options.Strict then succeed environment
-    else fail StrictModeDetected
+let simplify interactive environment = rop {
+    let! lockFile = environment |> PaketEnv.ensureLockFileExists
 
-let simplify interactive environment =
-    match environment.LockFile with
-    | Some lockFile ->
-        let flatLookup = getFlatLookup lockFile
-        let dependenciesFile = simplifyDependenciesFile(environment.DependenciesFile, flatLookup, interactive)
-        let projectFiles, referencesFiles = List.unzip environment.Projects
+    let flatLookup = getFlatLookup lockFile
+    let! dependenciesFile = simplifyDependenciesFile(environment.DependenciesFile, flatLookup, interactive)
+    let projectFiles, referencesFiles = List.unzip environment.Projects
 
-        let referencesFiles' =
-            referencesFiles
-            |> List.map (fun refFile -> simplifyReferencesFile(refFile, flatLookup, interactive))
-            |> Rop.collect
+    let referencesFiles' =
+        referencesFiles
+        |> List.map (fun refFile -> simplifyReferencesFile(refFile, flatLookup, interactive))
+        |> Rop.collect
 
-        let projects = List.zip projectFiles <!> referencesFiles'
+    let! projects = List.zip projectFiles <!> referencesFiles'
```

Comment:
``referencesFiles'`` could be also bound with ``let!``, so that the ``<!>`` operator is not necessary here